services/tracer/StreamWriterOnSeggerRtt.cpp

```diff
@@ -0,0 +1,20 @@
+#include "services/tracer/StreamWriterOnSeggerRtt.hpp"
+#include "RTT/SEGGER_RTT.h"
+
+namespace services
+{
+    StreamWriterOnSeggerRtt::StreamWriterOnSeggerRtt(unsigned int bufferIndex)
+        : bufferIndex(bufferIndex)
+    {}
+
+    void StreamWriterOnSeggerRtt::Insert(infra::ConstByteRange range, infra::StreamErrorPolicy& errorPolicy)
+    {
+        range.shrink_from_back_to(Available());
+        SEGGER_RTT_Write(bufferIndex, range.cbegin(), range.size());
+    }
+
+    std::size_t StreamWriterOnSeggerRtt::Available() const
+    {
+        return SEGGER_RTT_GetAvailWriteSpace(bufferIndex);
```

Comment:
How much is this typically? Can I configure this somewhere?

Reply:
Default is 1k, and can be overriden from the SEGGER_RTT_Conf.h by preprocessor defines.

E.g. BUFFER_SIZE_UP